app/src/main/java/org/dharmaseed/android/PlaybackService.java

```diff
@@ -169,6 +196,45 @@ public MediaNotification createNotification(MediaSession mediaSession, Immutable
 
     private class SessionCallback implements MediaSession.Callback {
 
+        @OptIn(markerClass = UnstableApi.class) @Override
+        public MediaSession.ConnectionResult onConnect(
+                MediaSession session, MediaSession.ControllerInfo controller) {
+
+            // Add seek forward/back commands to media notification
+            SessionCommands sessionCommands = new SessionCommands.Builder()
+                    .add(new SessionCommand(COMMAND_SEEK_BACK, new Bundle()))
+                    .add(new SessionCommand(COMMAND_SEEK_FORWARD, new Bundle()))
+                    .build();
+
+            // Remove default seek-to-previous commands from media notification
+            Player.Commands playerCommands = MediaSession.ConnectionResult.DEFAULT_PLAYER_COMMANDS
+                    .buildUpon()
+                    .removeAll(COMMAND_SEEK_TO_PREVIOUS, COMMAND_SEEK_TO_PREVIOUS_MEDIA_ITEM)
+                    .build();
+
+            return new MediaSession.ConnectionResult.AcceptedResultBuilder(session)
+                    .setAvailablePlayerCommands(playerCommands)
+                    .setAvailableSessionCommands(sessionCommands)
+                    .build();
+        }
+
+        @Override
+        public ListenableFuture<SessionResult> onCustomCommand(MediaSession session,
+                MediaSession.ControllerInfo controller, SessionCommand customCommand, Bundle args) {
+
+            final SessionResult result;
+            if (customCommand.customAction.equals(COMMAND_SEEK_BACK)) {
+                session.getPlayer().seekBack();
+                result = new SessionResult(SessionResult.RESULT_SUCCESS);
+            } else if (customCommand.customAction.equals(COMMAND_SEEK_FORWARD)) {
+                session.getPlayer().seekForward();
+                result = new SessionResult(SessionResult.RESULT_SUCCESS);
+            } else {
+                result = new SessionResult(SessionResult.RESULT_ERROR_NOT_SUPPORTED);
+            }
+            return Futures.immediateFuture(result);
```

Comment:
This all looks great to me. It's been a while since I dove into the media player code, but I don't see anything unexpected or problematic here.